test/common

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+declare -A COLOR
+COLOR['END']='\033[0m'
+COLOR['RED']='\033[0;91m'
+COLOR['GREEN']='\033[0;92m'
+COLOR['YELLOW']='\033[0;93m'
+COLOR['BLUE']='\033[0;94m'
+COLOR['WHITE']='\033[0;37m'
+COLOR['NC']='\033[0m'
+COLOR['CYAN']='\033[0;96m'
+
+readonly SKIP=200
+app=""
+stderr_log=""
+
+printc() {
+    local color="$1"
+    shift
+    echo -e "${COLOR[$color]}$@${COLOR['END']}"
+}
+
+check_root(){
+	if [ $USER != "root" ]; then
+		printc "RED" "Please run as root"
+    		exit 1
+	fi
+}
+
+check_output(){
+	printc "WHITE" "Command output:\n${APP_STDOUT}"
+	for regex in "${app_regex[@]}"; do
+		if echo "${APP_STDOUT}" | grep -q -P "$regex"; then
+			printc "BLUE" "[OK] Pattern '$regex' matched"
+			continue
+		fi
+		printc "YELLOW" "Output doesn't match: $regex"
+		return -1
+	done
+}
+
+test_init(){
+	app="${1#rt-}"
+	printc "BLUE" "\nTesting: " $app
+}
+
+test_failed() {
+	printc "RED" "FAIL"
+	exit 1
+}
+
+nonzero_fail() {
+	printc "RED" "FAIL - Application returned nonzero status"
+	printc "YELLOW" "[INFO] Command output:\n${APP_STDOUT}"
+	printc "RED" "[WARN] stderr:\n${APP_STDERR}"
+	exit 1
+}
+
+partial_success(){
+	printc "YELLOW" "[WARN] stderr:"
+	printc "YELLOW" ${APP_STDERR}
+	printc "NC" "[INFO] Command executed successfully"
+}
+
+app_exit_check(){
+	[$1 -ne 0] && nonzero_fail
+	partial_success
+}
+
+test_sucess() {
+	printc "GREEN" "PASS"
+	exit 0
+}
+
+# Catch the stout and stderr to different variables
```

Comment:
```suggestion
# Catch the stdout and stderr to different variables
```

Reply:
Doesn't this function also execute the program?

Reply:
@farshidtz yes! It's being executed in `"${@}";` part.

Reply:
So, the comment should be adjusted, explaining this.